wfe2/wfe.go

```diff
@@ -2377,6 +2378,71 @@ func (wfe *WebFrontEndImpl) RenewalInfo(ctx context.Context, logEvent *web.Reque
 		time.Unix(0, cert.Expires).UTC()))
 }
 
+// UpdateRenewal is used by the client to inform the server that they have
+// replaced the certificate in question, so it can be safely revoked. All
+// requests must be authenticated to the account which ordered the cert.
+func (wfe *WebFrontEndImpl) UpdateRenewal(ctx context.Context, logEvent *web.RequestEvent, response http.ResponseWriter, request *http.Request) {
+	if !features.Enabled(features.ServeRenewalInfo) {
+		wfe.sendError(response, logEvent, probs.NotFound("Feature not enabled"), nil)
+		return
+	}
+
+	body, _, acct, prob := wfe.validPOSTForAccount(request, ctx, logEvent)
+	addRequesterHeader(response, logEvent.Requester)
+	if prob != nil {
+		// validPOSTForAccount handles its own setting of logEvent.Errors
+		wfe.sendError(response, logEvent, prob, nil)
+		return
+	}
+
+	var updateRenewalRequest struct {
+		CertID   string `json:"certID"`
+		Replaced bool   `json:"replaced"`
+	}
+	err := json.Unmarshal(body, &updateRenewalRequest)
+	if err != nil {
+		wfe.sendError(response, logEvent, probs.Malformed("Unable to unmarshal RenewalInfo POST request body"), err)
+		return
+	}
+
+	der, err := base64.RawURLEncoding.DecodeString(updateRenewalRequest.CertID)
```

Comment:
If I send padded base64 data to this endpoint, it'll output error "illegal base64 data at input byte XYZ". Should we strip the padding to assist clients incorrectly sending it? The [draft RFC](https://datatracker.ietf.org/doc/html/draft-aaron-acme-ari-03#section-4.2) states, emphasis mine:
> certID (required, string): The CertID of the certificate whose renewal information should be updated, in the base64url-encoded version of the DER format with **trailing "=" stripped**. Note: this is identical to the final path component constructed for GET requests above.

https://go.dev/play/p/ikjseZtOo4M

If the decision is to keep it as is, perhaps we should return a more meaningful error for that case like, "certID should not contain base64url padding."